server/container/guice/guice-common/src/main/java/org/apache/james/utils/DataProbeImpl.java

```diff
@@ -52,9 +53,14 @@ private DataProbeImpl(
 
     @Override
     public void addUser(String userName, String password) throws Exception {
+        validateUsername(userName);
         usersRepository.addUser(userName, password);
     }
 
+    private void validateUsername(String userName) {
```

Comment:
If it throw when failing, I think I will prefer "assertValideUsername".
Moreover I think using exception instead of boolean for checking user input is not really good.